src/layout/dynamic_layouts/edpc_layout.cpp

```diff
@@ -0,0 +1,128 @@
+                            for (unsigned int l=j-1; l<j+2; l++) {
+                                grid[k][l] = AsciiLayoutSpec::CellType::DistillationRegion_0;
+                            }
+                        }
+                    }
+                }
+                else if (grid[i-2][j] == 'Q') {
+                    if (grid[i+1][j-1] == 'r') {
+                        grid[i][j] = AsciiLayoutSpec::CellType::ReservedForMagicState;                      
+                        for (unsigned int k=i+1; k<i+t_distillation_region_rows+1; k++) {
+                            for (unsigned int l=j-1; l<j+2; l++) {
+                                grid[k][l] = AsciiLayoutSpec::CellType::DistillationRegion_0;
+                            }
+                        }
+                    }
+                }
+                else if (grid[i][j+2] == 'Q') {
+                    if (grid[i-1][j-1] == 'r') {
+                        grid[i][j] = AsciiLayoutSpec::CellType::ReservedForMagicState;
+                        for (unsigned int k=i-1; k<i+2; k++) {
+                            for (unsigned int l=j-t_distillation_region_cols; l<j; l++) {
+                                grid[k][l] = AsciiLayoutSpec::CellType::DistillationRegion_0;
+                            }
+                        }
+                    }
+                }
+                else if (grid[i][j-2] == 'Q') {
+                    if (grid[i-1][j+1] == 'r') {
+                        grid[i][j] = AsciiLayoutSpec::CellType::ReservedForMagicState;
+                        for (unsigned int k=i-1; k<i+2; k++) {
+                            for (unsigned int l=j+1; l<j+t_distillation_region_cols+1; l++) {
+                                grid[k][l] = AsciiLayoutSpec::CellType::DistillationRegion_0;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    // Add dead cells
+    for (unsigned int i = 0; i<grid.size(); i++) {
+        for (unsigned int j = 0; j<grid[i].size(); j++) {
+            if (i >= t_distillation_region_rows && i < grid.size() - t_distillation_region_rows && j >= t_distillation_region_cols && j < grid[i].size() - t_distillation_region_cols) {
+                continue;
+            }
+            else if (grid[i][j] == 'r') {
+                grid[i][j] = AsciiLayoutSpec::CellType::DeadCell;
+            }
+        }
+    }
+
+
+    // Print out the grid for the purposes of debugging
+    // std::cout << "The number of rows in the grid is: " << grid.size() << std::endl;
+    // std::cout << "The number of columns in the grid is: " << grid[grid.size()-1].size() << std::endl;
+    // for (unsigned int i=0; i<grid.size(); i++) {
+    //     for (unsigned int j=0; j<grid[i].size(); j++) {
+    //         std::cout << grid[i][j] << ' ';
+    //     }
+    //     std::cout << std::endl;
+    // }
```

Comment:
Could do something like this:

```suggestion
#ifdef DEBUG_EDPC_LAYOUT_CREATION
    // Print out the grid for the purposes of debugging
    // std::cout << "The number of rows in the grid is: " << grid.size() << std::endl;
    // std::cout << "The number of columns in the grid is: " << grid[grid.size()-1].size() << std::endl;
    // for (unsigned int i=0; i<grid.size(); i++) {
    //     for (unsigned int j=0; j<grid[i].size(); j++) {
    //         std::cout << grid[i][j] << ' ';
    //     }
    //     std::cout << std::endl;
    // }
#endif
```

Reply:
Made this change in the newest commit.